File: Formação Lógica de Programação/JavaScript/Geração de Biomas em um Mundo de Blocos.js
const quantidadeGolpes = 4; // No desafio será: parseInt(gets());

let minerais = ["Carvao", "Ferro", "Diamante", "Pedra"];

for (let i = 1; i <= quantidadeGolpes; i++) {
  let minaIndex = (i - 1) % minerais.length;
  
  console.log(i + ": " + minerais[minaIndex]);
}

File: Formação Lógica de Programação/Python/Geração de Biomas em um Mundo de Blocos.py
quantidade_golpes = 4  # No desafio será: int(input())

minerais = ["Carvao", "Ferro", "Diamante", "Pedra"]

for i in range(1, quantidade_golpes + 1):
    # len(minerais) retorna o tamanho da lista, equivalente ao .length do JS
    mina_index = (i - 1) % len(minerais)
    
    # Utilizando f-strings para formatar o texto impresso no console
    print(f"{i}: {minerais[mina_index]}")